clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -995,6 +996,7 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
         MPM.addPass(ESIMDVerifierPass(LangOpts.SYCLESIMDForceStatelessMem));
         if (Level == OptimizationLevel::O0)
           MPM.addPass(ESIMDRemoveOptnoneNoinlinePass());
+        MPM.addPass(SYCLConditionalCallOnDevicePass(LangOpts.SYCLUniquePrefix));
```

Comment:
Design doc specifically says that this pass should be run before `SYCLPropagateAspectsUsagePass`, which is done in the code, but I think that it worth adding a comment into the code as well to make sure that it is still the case after any possible refactoring.